### ISSUE TITLE
Why the used memory of GPU is lower when warmup

### ISSUE THREAD

**Issue by Coobiw** [ultralytics/yolov5#8657]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello,It's my honor to push this issue here! When I train the model on my dataset with the setting as this:

`python train.py --workers 8 --device 0 --batch-size 16 --data data/data.yaml --img 960 960 --cfg cfg/training/yolov7.yaml --weights 'yolov7.pt' --adam`

And then, when I look up the log, I find that the used momery of GPU is lower during the first 4 epochs. After that, it keeps a stable value. So I really want to know what happen when warmup and how it leads to this outcome!

 Thanks for your reply!

**Comment by glenn-jocher** [comment]
@HustQBW CUDA memory usage should be stable after the first epoch. It will jump a bit after training on epoch 0 as the EMA is validated but afterwards for all other epochs memory usage should stay constant. Warmup does not have an effect on memory usage that I'm aware of, i.e.:

<img width="400" alt="Screen Shot 2022-07-23 at 7 38 29 PM" src="https://user-images.githubusercontent.com/26833433/180616570-dc2f1a7e-72f3-49a3-b618-f3cbd4f30155.png">

**Comment by Coobiw** [comment]
Thanks for your answer. I assume that I cant get you. It's embarrassing that I have another question.

Why the EMA is used during all periods when training. In my opinion, it may work well when the loss curve is jittering slightly(usually happening in the final stages of training.

**Comment by glenn-jocher** [comment]
@HustQBW EMA helps in general for most tasks and datasets. You can effectively turn off the EMA by setting `decay=0.0` here:
https://github.com/ultralytics/yolov5/blob/916bdb1d61f23de92833bd491df54cda5c3ef0cc/utils/torch_utils.py#L338

**Comment by Coobiw** [comment]
Really thanks for your help！

**Comment by Coobiw** [comment]
Actually,If possible,I also want to know if the EMA accelerate the coveragence when training on COCO dataset. Really thanks!

**Comment by Coobiw** [comment]
@glenn-jocher Hello! I have read train.py again and the EMA. In my opinion, EMA is in the whole time of training,if:
`            if ni - last_opt_step >= accumulate:
                scaler.step(optimizer)  # optimizer.step
                scaler.update()
                optimizer.zero_grad()
                if ema:
                    ema.update(model)
                last_opt_step = ni` 

So why the gpu momery usage in the first epoch is lower?

Thanks for your answer!

**Comment by glenn-jocher** [comment]
@HustQBW CUDA memory should be flat during training as shown above
